ran-core/src/main/java/io/ran/Property.java

```diff
@@ -344,4 +383,235 @@ public Property<T> getProperty() {
 			return property;
 		}
 	}
+
+	public abstract static class NumberValue<V extends Number> {
+		private static Map<Class, Function<Object, NumberValue>> types = new HashMap<>();
+		static {
+			types.put(Byte.class, ByteNumberValue::new);
+			types.put(byte.class, ByteNumberValue::new);
+			types.put(Short.class, ShortNumberValue::new);
+			types.put(short.class, ShortNumberValue::new);
+			types.put(Integer.class, IntegerNumberValue::new);
+			types.put(int.class, IntegerNumberValue::new);
+			types.put(Long.class, LongNumberValue::new);
+			types.put(long.class, LongNumberValue::new);
+			types.put(Float.class, FloatNumberValue::new);
+			types.put(float.class, FloatNumberValue::new);
+			types.put(Double.class, DoubleNumberValue::new);
+			types.put(double.class, DoubleNumberValue::new);
+			types.put(BigDecimal.class, BigDecimalNumberValue::new);
+		}
+		private final Object value;
+
+		public static <V extends Number, NV extends NumberValue<V>> NV get(V value) {
+			if (!types.containsKey(value.getClass())) {
+				return null;
+			}
+			return (NV) types.get(value.getClass()).apply(value);
+		}
+
+		public static <V extends Number, NV extends NumberValue<V>> NV get(Class<V> ofType, Object value) {
+			if (!types.containsKey(value.getClass())) {
+				return null;
+			}
+			return (NV) types.get(ofType).apply(value);
+		}
+
+		private NumberValue(Object value) {
+			this.value = value;
+		}
+
+		public V value() {
+			return (V) value;
+		}
+		public abstract NumberValue<V> add(NumberValue<?> other);
+
+		public abstract NumberValue<V> subtract(NumberValue<?> other);
+
+		public abstract NumberValue<V> multiply(NumberValue<?> numberValue);
+
+		public abstract NumberValue<V> divide(NumberValue<?> numberValue);
+
+		private static class ByteNumberValue extends NumberValue<Byte> {
+			public ByteNumberValue(Object value) {
+				super(((Number)value).byteValue());
+			}
+
+			@Override
+			public NumberValue<Byte> add(NumberValue<?> other) {
+				return new ByteNumberValue(value() + PropertyValue.getNumberValue(Byte.class,other.value()).value());
```

Comment:
```suggestion
				return new ByteNumberValue(value() + PropertyValue.getNumberValue(Byte.class, other.value()).value());
```
Nitpick, apply everywhere.